src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,425 @@
+  # - is_ring: Semantic flag indicating this linestring is part of a polygon structure
+  # - has_equal_ends: WKB format flag indicating if writer included duplicate closing point
+  if is_ring || has_equal_ends
+    # Return a Ring, exclude duplicate last point if present
+    if has_equal_ends
+      return Ring(points[1:end-1]...)
+    else
+      return Ring(points...)
+    end
+  else
+    # It's an open linestring, return a Rope
+    return Rope(points...)
+  end
+end
+
+function parse_wkb_polygon(io::IOBuffer, parse_coords::Function, crs_type)
+  num_rings = Int(ltoh(read(io, UInt32)))
+  
+  # Parse exterior ring using ring parsing function
+  exterior_ring = parse_wkb_linestring(io, parse_coords, crs_type; is_ring=true)
+  
+  # Parse interior rings (holes) using map-do syntax
+  hole_rings = map(2:num_rings) do ring_idx
+    parse_wkb_linestring(io, parse_coords, crs_type; is_ring=true)
+  end
+  
+  return PolyArea(exterior_ring, hole_rings...)
+end
+
+
+function parse_wkb_multipoint(io::IOBuffer, parse_coords::Function, crs_type)
+  num_points = Int(ltoh(read(io, UInt32)))
+  ndims = CoordRefSystems.ncoords(crs_type)
+  
+  points = map(1:num_points) do _
+    # Read sub-geometry header
+    byte_order_inner = read(io, UInt8)
+    wkb_type_inner = (byte_order_inner == 0x01 ? ltoh : ntoh)(read(io, UInt32))
+    
+    # Create parse_coords closure for this sub-geometry
+    read_value_inner = byte_order_inner == 0x01 ? ltoh : ntoh
+    parse_coords_inner = io -> read_coords_from_buffer(io, read_value_inner, ndims)
+    
+    parse_wkb_point(io, parse_coords_inner, crs_type)
+  end
+  
+  return Multi(points)
+end
+
+function parse_wkb_multilinestring(io::IOBuffer, parse_coords::Function, crs_type)
+  num_lines = Int(ltoh(read(io, UInt32)))
+  ndims = CoordRefSystems.ncoords(crs_type)
+  
+  lines = map(1:num_lines) do _
+    # Read sub-geometry header
+    byte_order_inner = read(io, UInt8)
+    wkb_type_inner = (byte_order_inner == 0x01 ? ltoh : ntoh)(read(io, UInt32))
+    
+    # Create parse_coords closure for this sub-geometry
+    read_value_inner = byte_order_inner == 0x01 ? ltoh : ntoh
+    parse_coords_inner = io -> read_coords_from_buffer(io, read_value_inner, ndims)
+    
+    parse_wkb_linestring(io, parse_coords_inner, crs_type)
+  end
+  
+  return Multi(lines)
+end
+
+function parse_wkb_multipolygon(io::IOBuffer, parse_coords::Function, crs_type)
+  num_polygons = Int(ltoh(read(io, UInt32)))
+  ndims = CoordRefSystems.ncoords(crs_type)
+  
+  polygons = map(1:num_polygons) do _
+    # Read sub-geometry header
+    byte_order_inner = read(io, UInt8)
+    wkb_type_inner = (byte_order_inner == 0x01 ? ltoh : ntoh)(read(io, UInt32))
+    
+    # Create parse_coords closure for this sub-geometry
+    read_value_inner = byte_order_inner == 0x01 ? ltoh : ntoh
+    parse_coords_inner = io -> read_coords_from_buffer(io, read_value_inner, ndims)
+    
+    parse_wkb_polygon(io, parse_coords_inner, crs_type)
+  end
```

Comment:
Why??? Why we are undoing the work we did already that used `parse_coords`? What am I missing?